source/byte_buf.c

```diff
@@ -146,9 +145,9 @@ int aws_byte_buf_init_copy_from_cursor(
     struct aws_byte_buf *dest,
     struct aws_allocator *allocator,
     struct aws_byte_cursor src) {
-    if (!allocator || !dest || !aws_byte_cursor_is_valid(&src)) {
-        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
-    }
+    AWS_ERROR_PRECONDITION(allocator);
+    AWS_ERROR_PRECONDITION(dest);
```

Comment:
Same, `allocator` and `dest` should probably assert.

Reply:
Done